san/backtest.py

```diff
@@ -22,40 +22,40 @@ def __init__(self, returns:pd.Series, trades:pd.Series, lagged=True, transaction
             trades.iloc[0] = False
         self.strategy_returns = ((returns * percent_invested_per_trade) * trades)
         self.trades = trades
-        
-        self.nr_trades = 0
+
+        self.nr_trades = {'buy': [], 'sell': []}
         for i in range(1, len(trades)):
             if trades[i] != trades[i - 1]:
                 self.strategy_returns.iloc[i] -= transaction_cost
-                self.nr_trades += 1
+                if trades[i]:
+                    self.nr_trades['buy'].append(self.trades.index[i])
+                else:
+                    self.nr_trades['sell'].append(self.trades.index[i])
         if trades[-1]:  # include last day sell to make benchmark possible
-            self.nr_trades += 1
+            self.nr_trades['sell'].append(self.trades.index[i])
 
-        self.performance = ((self.strategy_returns * percent_invested_per_trade) * trades + 1).cumprod() - 1
+        self.performance = ((self.strategy_returns * percent_invested_per_trade) * self.trades + 1).cumprod() - 1
         self.benchmark = (returns + 1).cumprod() - 1
-        self.sharpe_ratio = (self.strategy_returns.mean() * 365) / (self.strategy_returns.std() * np.sqrt(365))
-        
-        running_value = np.array(self.performance)
-        running_value[0] = 0
-        a = np.argmax(np.maximum.accumulate(running_value) - running_value)  # end of the period
-        b = np.argmax(running_value[:a])  # start of period
-        self.maximum_drawdown = ((running_value[a] - running_value[b]) / running_value[b]) * 100  # Maximum Drawdown
-
-
-    def get_sharpe_ratio(self):
-        return round(self.sharpe_ratio, 2)
 
+    def get_sharpe_ratio(self, decimals=2):
+        sharpe_ratio = (self.strategy_returns.mean() * 365) / (self.strategy_returns.std() * np.sqrt(365))
+        return round(sharpe_ratio, decimals)
 
     def get_value_at_risk(self, percentile=5):
         sorted_rets = sorted(self.strategy_returns)
         var = np.percentile(sorted_rets, percentile)
         return round(var * 100, 2)
 
     def get_nr_trades(self):
-        return self.nr_trades
+        return len(self.nr_trades['sell']) + len(self.nr_trades['buy'])
 
-    def get_maximum_drawdown(self):
-        return round(self.maximum_drawdown, 2)
+    def get_maximum_drawdown(self, decimals=2):
+        running_value = np.array(self.performance)
+        running_value[0] = 0
+        end = np.argmax(np.maximum.accumulate(running_value) - running_value)  # end of the dropdown period
+        start = np.argmax(running_value[:end])  # start of the dropdown period
+        maximum_drawdown = running_value[start] - running_value[end]
```

Comment:
Should we try `(end - start) / start` here instead to get the maximal drop in percent?

Reply:
we didnt go to any conclusion yet but I really need updated sanpy tomorrow. I propose leave it as it is now and open a new PR for that. (so I will not use it tomorrow)